lib/src/index.js

```diff
@@ -372,6 +375,93 @@ function initialize(element) {
             detail: event.detail,
         };
 
+    }
+
+    function verifyClickCell(cell)
+    {
+        const mouseDownCell = context.mouseDownCell;
+        if (context.state.resizableColumn || context.state.resizableRow)
+            return false;
+        if (context.didReorder)
+            return false;
+        if (cell === null)
+            return false;
+        if (stringifyId(cell.columnId) !== stringifyId(mouseDownCell.columnId))
+            return false;
+        if (stringifyId(cell.rowId) !== stringifyId(mouseDownCell.rowId))
+            return false;
+
+        return true;
+    }
+
+    function buildClickEventProperties(event)
+    {
+        return {
+            ctrlKey: event.ctrlKey,
+            shiftKey: event.shiftKey,
+            button: event.button,
+            buttons: event.buttons,
+            detail: event.detail,
+        };
+
+    }
+
+    context.addEventListener(element, 'contextmenu', (event) => {
+        // Do not forward context menu events whilst editing cells
+        if (context.state.text)
+            return;
+        updateState(context);
+
+        const cell = context.state.hoveredCell;
+
+        if (!verifyClickCell(cell))
+            return;
+
+        const column = context.state.columnLookup.get(stringifyId(cell.columnId));
+        const row = context.state.rowLookup.get(stringifyId(cell.rowId));
+        const eventProperties = buildClickEventProperties(event);
+
+        if (column.type === 'DATA' && row.type === 'DATA') {
+            context.state.options.onCellContextMenu({ ...context.state.hoveredCell, ...eventProperties });
+        } else if (column.type === 'CUSTOM' || row.type === 'CUSTOM') {
+            context.state.options.onCustomCellContextMenu({ ...context.state.hoveredCell, ...eventProperties });
+        }
+    })
+
+    context.addEventListener(element, 'contextmenu', (event) => {
```

Comment:
This code seems to be duplicated.

This also causes the event to be fired twice on every click.